lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1743,8 +1751,16 @@ LogicalResult FIRRTLLowering::visitDecl(MemOp op) {
           IntegerType::get(op.getContext(), std::max((size_t)1, width));
       auto accesses = getAllFieldAccesses(op.getResult(i), field);
 
-      Value wire = createTmpWireOp(
-          portType, ("." + portName + "." + field + ".wire").str());
+      auto wireName = ("." + portName + "." + field + ".wire").str();
+      // If no access to the subfield, then the wire cannot be optimized away,
+      // attach a symbol to it. It cannot be optimized because no rd/wr to it,
+      // no source value to replace it with.
```

Comment:
Why?  synthetic wires with no read or write seem ideal to eliminate.

Reply:
Was trying to handle the following case, we can't remove this wire. 
But I realized we should just handle this case during optimization, if wire read op has users, we cannot remove the wire. I will remove this symbol, and address your other comment in canonicalization.  
``` %.rw.wdata.wire = sv.wire  : !hw.inout<i42>
    %0 = sv.read_inout %.rw.wdata.wire : !hw.inout<i42>
    %_M.ro_data_0, %_M.rw_rdata_0 = hw.instance "_M" @FIRRTLMem_1_1_1_42_12_0_1_0(%clock1, %true, %c0_i4, %clock1, %true, %c0_i4, %true, %true, %0, %clock2, %inpred, %c0_i4, %true, %indata) : (i1, i1, i4, i1,
i1, i4, i1, i1, i42, i1, i1, i4, i1, i42) -> (i42, i42)```